README.md

```diff
@@ -32,3 +33,119 @@ async def send_hello_world():
     print(f'Sent push notification with ID {apns_id}')
     await client.close()
 ```
+
+### On idempotency
```

Comment:
all these added things should probably be moved to `docs/`. They go way too nitty gritty for a readme but would make great additions to the reference docs.

Reply:
Done!